### PR TITLE
Misc windows compatibility

### PR DESCRIPTION
**Pull request by krocard**
Several changes for windows compatibility:
 - do not add the -fPIC flac when compiling on windows
 - change libxml2 log callback in order not to need to format the message withm sprintf.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/171%23discussion_r38073137%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/171%23discussion_r38073169%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/171%23issuecomment-135341165%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/171%23issuecomment-135356749%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/171%23issuecomment-135370331%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/171%23issuecomment-135341165%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40OznOg%20%40dawagner%20please%20review%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A36%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-08-27T09%3A15%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22What%20about%20the%20%5C%22not%5C%22%20keyword%20%3F%22%2C%20%22created_at%22%3A%20%222015-08-27T10%3A07%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20cababc2a0637709315a26e06339bd643bdf823c4%20xmlserializer/XmlSerializingContext.h%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/171%23discussion_r38073137%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22remove%20line%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A34%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-08-27T08%3A35%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20xmlserializer/XmlSerializingContext.h%3AL53-67%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/171#issuecomment-135341165'>General Comment</a></b>
- <a href='https://github.com/krocard'></a> @OznOg @dawagner please review
- <a href='https://github.com/OznOg'></a> :+1:
- <a href='https://github.com/dawagner'></a> What about the "not" keyword ?
- [x] <a href='#crh-comment-Pull cababc2a0637709315a26e06339bd643bdf823c4 xmlserializer/XmlSerializingContext.h 14'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/171#discussion_r38073137'>File: xmlserializer/XmlSerializingContext.h:L53-67</a></b>
- <a href='https://github.com/krocard'></a> remove line

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/171?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/171?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/171'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/171'>Refresh</a>.<a href='#crh-update-none'></a>

File: xmlserializer/XmlSerializingContext.h [hunk 1]
```diff
@@ -50,13 +53,14 @@ class CXmlSerializingContext
     // Error
     void setError(const std::string& strError);
     void appendLineToError(const std::string& strAppend);
+
```
Comment (krocard) [inline xmlserializer/XmlSerializingContext.h]: remove line

Comment (krocard) [inline xmlserializer/XmlSerializingContext.h]: :+1:

**Comment by krocard** [comment]
@OznOg @dawagner please review

**Comment by OznOg** [comment]
:+1:

**Comment by dawagner** [comment]
What about the "not" keyword ?

**Pull request merged**